src/background.js

```diff
@@ -422,47 +422,33 @@ const rollout = {
     let skipHeuristicsCheck = await rollout.getSetting(DOH_SKIP_HEURISTICS_PREF, false);
     log("skipHeuristicsCheck: ", skipHeuristicsCheck);
 
-    await rollout.runStartupHeuristics();
-
-
+    await rollout.runHeuristics("startup");
```

Comment:
This is not really correct:
 First (before this line) add  browser.experiments.netChange.onConnectionChanged listener,
Second add browser.captivePortal.onConnectivityAvailable listener.
The two listeners are correct.
call  await rollout.runHeuristics("startup"); . 
The "startup" and "netChange" needs to be removed, they need to be detected in a different way.

runHeuristics function should check for network and captive portal state before doing anything and if If nsINetworkLinkService.linkStatusKnown or nsINetworkLinkService.isLinkUp is false or browser.captivePortal.getState() is not  neither "unlocked_portal" nor "not_captive exit without running heuristics.  We have setup listeners for the network-change events and the captive portal events and they will trigger a new runHeuristics.

This code may run multiple heuristics in parallel. That is fine, but we need to make sure to disable/enable TRR only when the results of the runHeuristics that is triggered the last are ready and ignore results of other runHeuristics